contracts/RecoveryManager.sol

```diff
@@ -0,0 +1,88 @@
+pragma solidity 0.4.8;
+
+import "./IdentityManager.sol";
+import "./Proxy.sol";
+
+contract RecoveryManager {
+    IdentityManager identityManager;
+    
+    struct Delegate {
+        bool exists;
+        address proposedUserKey;
+    }
+    
+    struct RecoveryIdentity {
+        bool exists;
+        uint neededSignatures;
+        mapping (address => Delegate)  delegates;
+        address[] delegateAddresses; // needed for iteration of mapping
+    }
+    
+    //Create RecoveryManager w/ partner _identityManager
+    function RecoveryManager (address _identityManager) {
+        identityManager = IdentityManager(_identityManager);
```

Comment:
I don't know how complex it would make it and I like simple in this. But can we somehow decouple it from the specific IdentityManager instance. In other words have it be part of the `RecoveryIdentity`.